mkosi/__init__.py

```diff
@@ -3485,9 +3492,13 @@ def make_cpio(state: MkosiState) -> Optional[BinaryIO]:
 
         compressor = compressor_command(should_compress_output(state.config))
         files = find_files(root_dir)
-        cmd: List[PathString] = [
-            "cpio", "-o", "--reproducible", "--null", "-H", "newc", "--quiet", "-D", root_dir
-        ]
+        cmd: List[PathString] = [cpio_binary(), "-o", "--null", "-H", "newc", "--quiet"]
+        if cpio_binary() == "bsdcpio":
+            cmd += ["-d"]
```

Comment:
GNU cpio has the same option. Is there a reason to add it here? Maybe the same would apply for GNU cpio as well.

Reply:
Well originally it was "-D" in this code, while man bsdcpio says "-d". I checked man gnutar and it says "-d".

Reply:
Well, now I'm confused. `-d` and `-D` do different things in GNU cpio. `-D` changes the working directory and `-d` "creates leading directories where needed". BSD cpio seems to support only the latter. 

My questions was that if creating needed directories was needed for BSD cpio, might it also be needed or beneficial in the GNU case. Might your issue #1206 stem from BSD cpio not supporting changing the directory?